test/README.md

```diff
@@ -1,95 +1,296 @@
+
+### One time setup
+
+#### Setting up your local machine
+
+##### Prerequisites
+To run the End-to-End tests, we will be building the required binaries from the code, build container images and push them to a local docker repository. The tests will install the IoT Edge runtime from the binaries on your machine and run the containers using your local repository.
+
+It is important that your machine satisfies the requirements to run IoT Edge. See our installation docs ([Linux](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge-linux), [Windows](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge-windows)) for more information on prerequisites. In particular, make sure to prepare your machine to access the Microsoft installation pacakges. *The steps to do this are listed below for your convenience. Again, please look at the section in the linked document for your os/arch combination if it is different from ubuntu18.04/amd64*
+~~~sh
+# Setup the repository information
+curl https://packages.microsoft.com/config/ubuntu/18.04/multiarch/prod.list > ./microsoft-prod.list
+
+# Setup the Microsoft GPG pulic key in the apt's trusted list.
+sudo cp ./microsoft-prod.list /etc/apt/sources.list.d/
+curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.gpg
+sudo cp ./microsoft.gpg /etc/apt/trusted.gpg.d/
+
+# Run update to update the package lists on your machine
+sudo apt-get update
+~~~
+
+Also, create a rootCA certificate for your machine *(See [Create Certs](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-create-test-certificates?view=iotedge-2020-11#create-root-ca-certificate) for more details)*. Navigate to the directory where the certificate generation scripts are found (typically iotedge/tools/CACertificates) and run
+~~~ sh
+./certGen.sh create_root_and_intermediate
+
+# This will create root and intermediate CA certs. The Root CA cert will be located in the 
+# certs sub directory and will be needed to run the tests.
+~~~
+
+##### Install java
+The build scripts use Java for code generation tasks, so you will need a jdk on your machine.
+Here is how you can install the openjdk.
+~~~ sh
+sudo apt-get install openjdk-11-jdk
+~~~
+
+##### Configure your local container repository
+We will be building the container images from the codebase and pushing them to  local docker repository. To prepare for this, we will install docker and configure the docker repository using the steps below. See [Deploy a registry server](https://docs.docker.com/registry/deploying/) for further information regarding these steps to configure the registry.
+
+###### Install docker
+See [Install docker using the convenience scripts](https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script) for details
+
+~~~ sh
+curl -fsSL https://get.docker.com -o get-docker.sh
+sudo sh get-docker.sh 
+~~~

```

Comment:
Should we instead encourage install of the Moby engine? https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge?view=iotedge-2020-11#install-a-container-engine

Reply:
Good callout. Done.